src/transformState.js

```diff
@@ -4,8 +4,37 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
+/* - `state` is an object.
+- actions is an array of objects
+type contains  `'addProperties'` `'removeProperties'` `'clear'`;
+- if `type` is `addProperties`, second property is `extraData`.
+- if `type` is `removeProperties`, second property is `keysToRemove`
+- if `type` is `clear` you should remove all the properties
+ Use `switch
+if key is a variable - `object[key]`.
+If it's called key -  `object.key``
+*/
+
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete state[key]);
+        break;
+
+      case 'clear':
+        Object.keys(state).forEach((key) => delete state[key]);
```

Comment:
It's one of the ways to delete all properties from an object. The other one is to assign an empty object to the `state` variable. In the second case, you'll create a new reference so they're not identical.

The first way is slower, creating a nested loop but keeping the same reference. The second way is faster, without a nested loop, but it creates a new reference.